pyvista/plotting/plotting.py

```diff
@@ -4073,6 +4081,13 @@ def show(self, title=None, window_size=None, interactive=True,
                                      height=height)
             except:
                 pass
+        # In the event that the user hit the exit-button ont the windows GUI,
+        # then it must be finalized and deleted as accessing it will kill the
+        # kernel. Here we check for that and clean it up before moving on to
+        # any of the closing routines that might try to still access that
+        # render window.
+        if not self.ren_win.IsCurrent():
+            self._clear_ren_win()
```

Comment:
I'm tempted to add a warning here if the user is on Windows saying:

> Please use the q-key to close the Plotting window when using a Windows computer as screenshots may not be properly captured when using the exit button

No issues on Mac/Linux, but Windows people will not be able to save screenshots correctly if they hit the exit button in a code snippet like:

```py
import pyvista as pv
mesh = pv.Sphere()
p = pv.Plotter(notebook=0)
p.add_mesh(mesh)
p.show(screenshot="foo.png")
```

The screenshot saved will be the first rendered frame as we can't capture a screenshot if the user destroys the OpenGL context when hitting the exit button

Reply:
Adding it in the docs (or under `Notes`) might be helpful for windows users as it would would at least show up in the docs and would be searchable.

Also, nice fix with getting it to work on Windows @banesullivan

Reply:
Added in Notes section